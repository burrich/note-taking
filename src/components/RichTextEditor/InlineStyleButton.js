import React from 'react';
import StyleButton from './StyleButton';

/**
 * InlineStyleButton component.
 */
function InlineStyleButton(props) {
  const style = styles[props.name];
  const active = props.editorStyle.has(style.code);
  
  return (
    <StyleButton 
      style={style}
      active={active}
      onToggle={props.onToggle}
      disabled={props.disabled} />
  );
}

// code for draft and name for semantic ui icons
const styles = {
  bold: {
    code: 'BOLD',
    name: 'bold'
  },
  italic: {
    code: 'ITALIC',
    name: 'italic'
  },
  underline: {
    code: 'UNDERLINE',
    name: 'underline'
  },
  strikethrough: {
    code: 'STRIKETHROUGH',
    name: 'strikethrough'
  },
}

export default InlineStyleButton;
