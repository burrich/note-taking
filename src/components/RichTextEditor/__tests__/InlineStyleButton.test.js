import React from 'react';
import { shallow } from 'enzyme';
import InlineStyleButton from '../InlineStyleButton';

const buttonStyle = {
  code: 'BOLD',
  name: 'bold'
};
const inlineStyle = new Map();
const toggleInlineStyle = jest.fn();

test('Inactive inline style button renders correctly', () => {
  const button = shallow(
    <InlineStyleButton 
      name={buttonStyle.name}
      editorStyle={inlineStyle}
      onToggle={toggleInlineStyle} />
  );
  expect(button).toMatchSnapshot();
});

test('Active inline style button renders correctly', () => {
  inlineStyle.set(buttonStyle.code, buttonStyle.code);

  const button = shallow(
    <InlineStyleButton 
      name={buttonStyle.name}
      editorStyle={inlineStyle}
      onToggle={toggleInlineStyle} />
  );
  expect(button).toMatchSnapshot();
});