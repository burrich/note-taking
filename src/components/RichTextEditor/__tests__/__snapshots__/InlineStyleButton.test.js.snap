// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Active inline style button renders correctly 1`] = `
<StyleButton
  active={true}
  onToggle={[Function]}
  style={
    Object {
      "code": "BOLD",
      "name": "bold",
    }
  }
/>
`;

exports[`Inactive inline style button renders correctly 1`] = `
<StyleButton
  active={false}
  onToggle={[Function]}
  style={
    Object {
      "code": "BOLD",
      "name": "bold",
    }
  }
/>
`;
